a3/terraform/modules/cluster/gke/main.tf

```diff
@@ -252,7 +252,7 @@ resource "google_container_node_pool" "node-pools" {
     for_each = var.node_pools[count.index].use_compact_placement_policy ? [1] : []
     content {
       type        = "COMPACT"
-      policy_name = "${var.resource_prefix}-np-${count.index}"
+      policy_name = module.resource_policy[count.index].resource_name
```

Comment:
have you tested this? `count.index` is an integer while `module.resource_policy` is a map that should be indexed by a string -- I'm surprised terraform would allow you to do this.

Reply:
right. terraform allows this but this logic would be wrong. since some node pools might not use resource policy.

Reply:
addressed.